File: src/utils/Logger.cpp
#include "utils/Logger.h"

void Logger::begin(unsigned long baudRate) {
    Serial.begin(baudRate);
    while (!Serial && millis() < 2000) {
        delay(10);
    }
}

void Logger::debug(const char* message) {
    log(DEBUG, message);
}

void Logger::info(const char* message) {
    log(INFO, message);
}

void Logger::warning(const char* message) {
    log(WARNING, message);
}

void Logger::error(const char* message) {
    log(ERROR, message);
}

void Logger::log(Level level, const char* message) {
    unsigned long timestamp = millis();
    
    Serial.print("[");
    Serial.print(timestamp);
    Serial.print("] ");
    Serial.print(getLevelString(level));
    Serial.print(": ");
    Serial.println(message);
}

const char* Logger::getLevelString(Level level) {
    switch (level) {
        case DEBUG: return "DEBUG";
        case INFO: return "INFO";
        case WARNING: return "WARN";
        case ERROR: return "ERROR";
        default: return "UNKNOWN";
    }
}

File: backend/src/utils/logger.js
const LOG_LEVEL = process.env.LOG_LEVEL || 'info';

const levels = {
  debug: 0,
  info: 1,
  warn: 2,
  error: 3
};

const colors = {
  reset: '\x1b[0m',
  debug: '\x1b[36m',
  info: '\x1b[32m',
  warn: '\x1b[33m',
  error: '\x1b[31m'
};

function log(level, message, data) {
  if (levels[level] < levels[LOG_LEVEL]) return;
  
  const timestamp = new Date().toISOString();
  const color = colors[level] || '';
  const reset = colors.reset;
  
  let output = `${color}[${timestamp}] ${level.toUpperCase()}:${reset} ${message}`;
  
  if (data) {
    output += ` ${JSON.stringify(data)}`;
  }
  
  console.log(output);
}

module.exports = {
  debug: (msg, data) => log('debug', msg, data),
  info: (msg, data) => log('info', msg, data),
  warn: (msg, data) => log('warn', msg, data),
  error: (msg, data) => log('error', msg, data)
};
